rest/route/aws.go

```diff
@@ -98,3 +117,110 @@ func (aws *awsSns) Run(ctx context.Context) gimlet.Responder {
 
 	return gimlet.NewJSONResponse(struct{}{})
 }
+
+type eventBridgeNotification struct {
+	DetailType string      `json:"detail-type"`
+	Detail     eventDetail `json:"detail"`
+}
+type eventDetail struct {
+	InstanceID string `json:"instance-id"`
+	State      string `json:"state"`
+}
+
+func (aws *awsSns) handleNotification(ctx context.Context) error {
+	notification := &eventBridgeNotification{}
+	if err := json.Unmarshal([]byte(aws.payload.Message), notification); err != nil {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    errors.Wrap(err, "problem unmarshalling notification").Error(),
+		}
+	}
+
+	switch notification.DetailType {
+	case interruptionWarningType:
+		if err := aws.handleInstanceInterruptionWarning(ctx, notification.Detail.InstanceID); err != nil {
+			return gimlet.ErrorResponse{
+				StatusCode: http.StatusInternalServerError,
+				Message:    errors.Wrap(err, "problem processing interruption warning").Error(),
+			}
+		}
+
+	case instanceStateChangeType:
+		if notification.Detail.State == ec2.InstanceStateNameTerminated {
+			if err := aws.handleInstanceTerminated(ctx, notification.Detail.InstanceID); err != nil {
+				return gimlet.ErrorResponse{
+					StatusCode: http.StatusInternalServerError,
+					Message:    errors.Wrap(err, "problem processing instance termination").Error(),
+				}
+			}
+		}
+
+	default:
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    fmt.Sprintf("unknown detail type '%s'", notification.DetailType),
+		}
+	}
+
+	return nil
+}
+
+func (aws *awsSns) handleInstanceInterruptionWarning(ctx context.Context, instanceID string) error {
+	h, err := aws.sc.FindHostById(instanceID)
+	if err != nil {
+		return err
+	}
+	// don't make AWS keep retrying if we haven't heard of the host
+	if h == nil {
```

Comment:
nit: it doesn't seem like the line is too long to include them all together (or at least the second two clauses, if you want to keep h == nil separate for the comment)
```
if h == nil || h.Status == evergreen.HostTerminated || skipEarlyTermination(h) { 
    return nil 
} 
```

Reply:
I prefer to keep them separate for clarity